src/clj/commiteth/routes/services.clj

```diff
@@ -214,26 +215,41 @@
                 (do
                   (log/debug "/usage-metrics" user)
                   (ok (usage-metrics/usage-metrics-by-day))))
+
            (context "/user" []
+
                     (GET "/" {:keys [params]}
                          :auth-rules authenticated?
                          :current-user user
                          (ok (handle-get-user user (:token params))))
-                    (POST "/address" []
+
+                    (POST "/" []
                           :auth-rules authenticated?
-                          :body-params [user-id :- Long, address :- String]
-                          :summary "Update user address"
-                          (if-not (eth/valid-address? address)
-                            (do
-                              (log/debug "POST /address: invalid input" address)
-                              {:status 400
-                               :body (str "Invalid Ethereum address '" address "'")})
-                            (let [result (users/update-user-address
-                                          user-id
-                                          address)]
-                              (if (= 1 result)
-                                (ok)
-                                (internal-server-error)))))
+                          :current-user user
+                          :body [body {(s/optional-key :address) s/Str
```

Comment:
Why could these params be optional?

Reply:
Because here, we may accept either an address or a hidden flag only instead of passing both of them at once. Say, when I press "update" my address, I send an address only. The same is for hidden flag.